src/pkm/PK3RR.ts

```diff
@@ -0,0 +1,371 @@
+  trainerFriendship: number
+  moves: number[]
+  movePP: number[]
+  evs: types.Stats
+  pokerusByte: number
+  metLocationIndex: number
+  metLevel: number
+  gameOfOrigin: number
+  ball: number
+  ivs: types.Stats
+  isEgg: boolean
+  isNicknamed: boolean
+  currentHP: number = 0;
+  nickname: string
+  trainerName: string
+  trainerGender: boolean
+
+  constructor(arg: ArrayBuffer | AllPKMFields, encrypted?: boolean) {
+    if (arg instanceof ArrayBuffer) {
+      let buffer = arg
+      const dataView = new DataView(buffer)
+      
+      // https://github.com/Skeli789/Complete-Fire-Red-Upgrade/blob/master/include/new/pokemon_storage_system.h
+
+      // Personality 0:4
+      this.personalityValue = dataView.getUint32(0x0, true)
+
+      // OTID 4:8
+      this.trainerID = dataView.getUint16(0x4, true)
+      this.secretID = dataView.getUint16(0x6, true)
+
+      // Nickname 8:18
+      this.nickname = stringLogic.readGen3StringFromBytes(dataView, 0x8, 10)
+
+      // Language 18
+      this.languageIndex = dataView.getUint8(0x12)
+
+      // Sanity 19
+      // this.sanity = dataView.getUint8(0x13)
+
+      // OT Name 20:27
+      this.trainerName = stringLogic.readGen3StringFromBytes(dataView, 0x14, 7)
+
+      // Markings 27
+      this.markings = types.markingsFourShapesFromBytes(dataView, 0x1b)
+      
+      // Species 28:30 !!!
+      this.dexNum = conversion.fromGen3PokemonIndex(dataView.getUint16(0x1c, true))
+      
+      // Held Item 30:32
+      this.heldItemIndex = dataView.getUint16(0x1e, true)
+
+      // Exp 32:36
+      this.exp = dataView.getUint32(0x20, true)
+      
+      // Move PP Up 36
+      this.movePPUps = [
+        byteLogic.uIntFromBufferBits(dataView, 0x24, 0, 2, true),
+        byteLogic.uIntFromBufferBits(dataView, 0x24, 2, 2, true),
+        byteLogic.uIntFromBufferBits(dataView, 0x24, 4, 2, true),
+        byteLogic.uIntFromBufferBits(dataView, 0x24, 6, 2, true),
+      ]
+
+      // Friendship 37
+      this.trainerFriendship = dataView.getUint8(0x25)
+
+      // Pokeball 38
+      this.ball = 0 // byteLogic.uIntFromBufferBits(dataView, 0x26, 11, 4, true)
+
+      // // Moves 38:43 (5 bytes total for 4 moves with 10 bits each)
+      // this.moves = [
+      //   byteLogic.uIntFromBufferBits(dataView, 0x27, 0, 10, true), // Move 1
+      //   byteLogic.uIntFromBufferBits(dataView, 0x27, 10, 10, true), // Move 2
+      //   byteLogic.uIntFromBufferBits(dataView, 0x27, 20, 10, true), // Move 3
+      //   byteLogic.uIntFromBufferBits(dataView, 0x27, 30, 10, true), // Move 4
+      // ]
+
+      // Moves
+      this.moves = [
+        // dataView.getUint16(0x2c, true),
+        // dataView.getUint16(0x2e, true),
+        // dataView.getUint16(0x30, true),
+        // dataView.getUint16(0x32, true),
+      ]
+      this.movePP = [
+        // dataView.getUint8(0x34),
+        // dataView.getUint8(0x35),
+        // dataView.getUint8(0x36),
+        // dataView.getUint8(0x37),
+      ]
```

Comment:
Did you mean to leave these commented out?

Reply:
That's so cool. I was worried you weren't a fan of Rom Hacks! I think they are such a cool way to explore the game we love in new and interesting ways.